### PR TITLE
Bug 1115622 - [Bluetooth] decouple bluetoothTransfer and NfcHandoverMana... +autoland

### PR DESCRIPTION
**Pull request by gasolin**
...ger's dependency

### CODE REVIEW

File: apps/system/js/bluetooth_transfer.js [hunk 1]
```diff
@@ -43,18 +36,28 @@ var BluetoothTransfer = {
...
-    window.addEventListener('bluetooth-opp-transfer-complete',
-      this.onTransferComplete.bind(this)
-    );
+    window.addEventListener('bluetooth-opp-transfer-complete', this);
+  },
+
+  handleEvent: function bt_handleEvent(evt) {
+    switch(evt.type) {
+      case 'iac-bluetoothTransfercomms':
+        this._onFilesSending();
```
Comment (ian-liu) [inline apps/system/js/bluetooth_transfer.js]: As above comment, missed to pass parameter to `this._onFilesSending()` function.

File: apps/system/js/bluetooth_transfer.js [hunk 2]
```diff
@@ -43,18 +36,28 @@ var BluetoothTransfer = {
...
+        break;
+      case 'bluetooth-opp-transfer-start':
+        this._onUpdateProgress('start');
```
Comment (ian-liu) [inline apps/system/js/bluetooth_transfer.js]: Same here.

File: apps/system/js/bluetooth_transfer.js [hunk 3]
```diff
@@ -43,18 +36,28 @@ var BluetoothTransfer = {
...
+        break;
+      case 'bluetooth-opp-transfer-complete':
+        this._onTransferComplete();
```
Comment (ian-liu) [inline apps/system/js/bluetooth_transfer.js]: Same here. I still like to keep original coding style. Please need to modify them beside of naming  for private methods. Thanks.

File: apps/system/js/bluetooth_transfer.js [hunk 4]
```diff
@@ -146,7 +149,7 @@ var BluetoothTransfer = {
     if (NfcHandoverManager.isHandoverInProgress()) {
       // Bypassing confirm dialog while incoming file transfer via NFC Handover
       this.debug('Incoming file via NFC Handover. Bypassing confirm dialog');
-      NfcHandoverManager.transferStarted();
+      window.dispatchEvent(new CustomEvent('nfc-transfer-started'));
```
Comment (ian-liu) [inline apps/system/js/bluetooth_transfer.js]: :100:

**Comment by try-server-hook** [comment]
[gasolin gasolin (gasolin)](https://github.com/gasolin) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=3933eed68410)

**Comment by try-server-hook** [comment]
[gasolin gasolin (gasolin)](https://github.com/gasolin) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=54362fbd6238)

**Comment by gasolin** [comment]
replaced by https://github.com/mozilla-b2g/gaia/pull/27140

**Pull request closed without merging**